dimagi/utils/dates.py

```diff
@@ -303,6 +306,11 @@ def get_validation_reason(self):
             return "You can't have an end date of %s after start date of %s" % (self.enddate, self.startdate)
         elif self.startdate < datetime.datetime(1900, 01, 01) or self.enddate < datetime.datetime(1900, 01, 01):
             return "You can't use dates earlier than the year 1900"
+        elif self.max_days >= 0:
+            delta = self.enddate - self.startdate
+            if delta.days > self.max_days:
+                return "You are limited to a span of {} days, but this date range spans {} days".format(
```

Comment:
if these are intended to be user facing i wonder if they should be translated (i see above the others aren't so not blocking)